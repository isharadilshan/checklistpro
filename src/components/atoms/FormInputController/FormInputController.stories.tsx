import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import FormInputController from './';

export default {
  title: 'components/atoms/FormInputController',
  component: FormInputController,
} as ComponentMeta<typeof FormInputController>;

export const Basic: ComponentStory<typeof FormInputController> = (args) => (
  <NativeBaseProvider>
    <FormInputController {...args} />
  </NativeBaseProvider>
);

Basic.args = {
  name: 'Form Input' || undefined,
  defaultValue: 'Default Value',
  error: 'ERROR',
  type: 'text',
  control: {},
};
