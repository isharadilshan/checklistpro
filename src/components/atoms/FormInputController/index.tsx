import {FormControl, Input} from 'native-base';
import React from 'react';
import {
  Control,
  Controller,
  FieldError,
  Path,
  RegisterOptions,
} from 'react-hook-form';
import createStyle from './styles';
interface FormInputControllerProps<FieldsType> {
  name: Path<FieldsType>;
  defaultValue?: string;
  rules?: RegisterOptions;
  error?: FieldError;
  control: Control<any>;
  type?: 'text' | 'password';
}

interface Props<FieldsType> extends FormInputControllerProps<FieldsType> {
  label?: string;
  placeholder?: string;
}

const FormInputController = <FieldsType,>({
  error,
  rules,
  label,
  control,
  name,
  placeholder,
  type = 'text',
}: Props<FieldsType>) => {
  const isInvalid = error != null;
  const isRequired = rules != null && 'required' in rules;

  const styles = createStyle();

  return (
    <FormControl
      isInvalid={isInvalid}
      isRequired={isRequired}
      style={styles.controlWrapper}
    >
      {label != null && <FormControl.Label>{label}</FormControl.Label>}
      <Controller
        control={control}
        name={name}
        rules={rules}
        render={({field: {onChange, onBlur, value}}) => (
          <Input
            size={'lg'}
            placeholder={placeholder}
            value={value}
            onBlur={onBlur}
            onChangeText={(val) => onChange(val)}
            style={{color: 'white'}}
            type={type}
          />
        )}
      />
      {error != null && (
        <FormControl.ErrorMessage>{error.message}</FormControl.ErrorMessage>
      )}
    </FormControl>
  );
};

export default FormInputController;
