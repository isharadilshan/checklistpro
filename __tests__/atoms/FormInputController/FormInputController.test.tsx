import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import FormInputController from '../../../src/components/atoms/FormInputController';

configure({adapter: new Adapter()});

const props = {
  name: 'TEST FORM INPUT CONTROLLER',
  defaultValue: 'test',
  error: 'This is not a valid form input',
  label: 'FORM INPUT',
};

describe('when rendering', () => {
  const wrapper = shallow(<FormInputController {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
