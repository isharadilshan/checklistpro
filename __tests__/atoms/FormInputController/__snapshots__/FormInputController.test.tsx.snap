// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Memo(ForwardRef(FormControl))
  isInvalid={true}
  isRequired={false}
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Memo(ForwardRef(FormControlLabel))>
    FORM INPUT
  </Memo(ForwardRef(FormControlLabel))>
  <Component
    name="TEST FORM INPUT CONTROLLER"
    render={[Function]}
  />
  <Memo(ForwardRef(FormControlErrorMessage)) />
</Memo(ForwardRef(FormControl))>
`;
